.github/scripts/generate_ci_workflows.py

```diff
@@ -193,7 +193,7 @@ def generate_workflow_file(self, workflow_template: jinja2.Template) -> None:
     ),
     CIWorkflow(
         arch="windows",
-        build_environment="win-vs2019-cuda11.1-py3",
+        build_environment="master-only-win-vs2019-cuda11.1-py3",
```

Comment:
We should probably add a way to do this automatically in the future instead of relying on the magic string that is `build_environment`

Reply:
yes. I had #62351 that originally had the idea to autogen, but I guess we losses some flexibility. Will try to address this in a separate PR.

Reply:
there's a potential change to move from master to main as the default branch, maybe better to use `trunk-only` to cover both master/main to avoid another naming change in the future.

Reply:
good point. let's make the change.